fastcrawler/test_utils.py

```diff
@@ -1,62 +1,73 @@
-from fastcrawler.engine import AioHttpEngine
+from enum import StrEnum
 
+from fastcrawler.engine.contracts import (
+    ProxySetting,
+    RequestCycle,
+    SetCookieParam,
+    Response,
+    Url,
+    Request,
+    EngineProto,
+)
 
-class TestServer:
-    def __init__(self):
-        self.responses = {
-            "GET": {"message": "GET request received"},
-            "POST": {"message": "POST request received"},
-            "PUT": {"message": "PUT request received"},
-            "DELETE": {"message": "DELETE request received"},
-        }
 
-    def get_response(self, method, data=None):
-        response = self.responses.get(method)
-        if data:
-            response["data"] = data
-        return response
+class Method(StrEnum):
+    GET = "GET"
+    POST = "POST"
+    PUT = "PUT"
+    DELETE = "DELETE"
 
 
-class TestClient(AioHttpEngine):
-    def __init__(self, server=TestServer(), headers=None, proxy=None):
-        self.server = server
+class MockEngine:
+    def __init__(
+        self,
+        cookies: list[SetCookieParam] | None = None,
+        headers: dict | None = None,
+        user_agent: str | None = None,
+        proxy: ProxySetting | None = None,
+        connection_limit: int | None = None,
+    ):
+        super().__init__(cookies, headers, user_agent, proxy, connection_limit)
 
-    def get(self, url):
-        response = self.server.get_response("GET")
-        return response
+    async def get(self, requests: list[Request]) -> dict[Url, RequestCycle]:
+        """GET HTTP Method for protocol to retrieve a list of URL."""
+        return await self.batch(requests, "GET")
 
-    def post(self, url, data):
-        response = self.server.get_response("POST", data=data)
-        return response
+    async def post(self, requests: list[Request]) -> dict[Url, RequestCycle]:
+        """POST HTTP Method for protocol to crawl a list of URL."""
+        return await self.batch(requests, "POST")
 
-    def delete(self, url):
-        response = self.server.get_response("DELETE")
-        return response
+    async def put(self, requests: list[Request]) -> dict[Url, RequestCycle]:
+        """PUT HTTP Method for protocol to crawl a list of URL."""
+        return await self.batch(requests, "PUT")
 
-    def put(self, url, data):
-        response = self.server.get_response("PUT", data=data)
-        return response
+    async def delete(self, requests: list[Request]) -> dict[Url, RequestCycle]:
+        """DELETE HTTP Method for protocol to crawl a list of URL."""
+        return await self.batch(requests, "DELETE")
```

Comment:
Now that's some great stuff! remember to include other method in contracts as well. there's one to translate response. if you don't mock everything, then your mock won't work. 
Remember you should also mock batch method, which I don't see in your code.